Google.Api.Generator/RoslynUtils/Keywords.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+
+namespace Google.Api.Generator.RoslynUtils
+{
+    internal static class Keywords
+    {
+        // Taken from: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/
+        private static IImmutableSet<string> s_keywords = ImmutableHashSet.Create(
```

Comment:
We could *potentially* include the contextual keywords as well - reason being that while having a parameter named `async` wouldn't hurt the compiler, it might confuse syntax highlighters.

Reply:
Yes I thought of this, but think I'd rather just leave them until we see a real case where it causes a problem.